From 2a843fe62d990851cf544214ace1be8e0c51e851 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Tue, 26 Nov 2019 10:52:50 +0100
Subject: [PATCH] Adapt to Moo-2.003006 that does not load
 Devel::GlobalDestruction
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Moo-2.003006 stopped loading Devel::GlobalDestruction on recent Perls.
As a result Object-Remote tests failed because:

$ perl -Ilib -e 'require Object::Remote::FatNode'
Use of uninitialized value $_ in hash element at lib/Object/Remote/FatNode.pm line 127.
Compilation failed in require at -e line 1.

Object::Remote::FatNode expected that Devel/GlobalDestruction.pm was
always presented in @INC and created an undefined entry in
@non_core_non_arch since Moo-2.003006.

This patch fixes it.

CPAN RT#130885

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 lib/Object/Remote/FatNode.pm | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/Object/Remote/FatNode.pm b/lib/Object/Remote/FatNode.pm
index 8b143b8..d0cf133 100644
--- a/lib/Object/Remote/FatNode.pm
+++ b/lib/Object/Remote/FatNode.pm
@@ -63,7 +63,8 @@ foreach(keys(%mods)) {
   }
 }
 
-my @non_core_non_arch = ( $file_names{'Devel/GlobalDestruction.pm'} );
+my @non_core_non_arch = exists $file_names{'Devel/GlobalDestruction.pm'} ?
+  ($file_names{'Devel/GlobalDestruction.pm'}) : ();
 push @non_core_non_arch, grep +(
   not (
     #some of the config variables can be empty which will eval as a matching regex
-- 
2.21.0

